test/fixtures/dialects/snowflake/snowflake_match_recognize.sql

```diff
@@ -18,7 +19,66 @@ select * from stock_price_history
   )
 order by company, match_number;
 
--- With more complicated pattern
+select price_date, match_number, msq, price, cl from
+  (select * from stock_price_history where company='ABCD') match_recognize(
```

Comment:
This was the query where match_recognize kept getting matched as an alias for the subquery. I tried adding it to some of the terminator grammars to no avail. its unlikely someone is using `match_recognize` as a column name so we should be pretty safe to just make match_recognize reserved for now